### DISCUSSION TITLE
How do I set a unique key for sunburst data?

### DISCUSSION THREAD

**Discussion by tavise** [plouc/nivo#2241] [Q&A]
(@nivo/sunburst: ^0.80.0)
 I have a ReactiveSunburst chart with the following data structure.  React puts large errors in the console: 
```
Warning: Encountered two children with the same key, `Foo`
```
I tried adding a key to each object, but that didn't work.  Anyone have ideas?  I'd prefer the tooltip for each slice to show "Foo" and not have to make that unique somewhow.
```
{
  "id": "Foo",
  "children": [
    {
      "id": "Foo",
      "children": []
    },
    {
      "id": "Bar",
      "children": []
    },
  ]
}
```

**Comment by plouc** [comment]
I should probably not use just the ID for hierarchical data, and concatenate it with ancestors' ids, which would make keys unique.

**Reply by piratefinn** [reply]
> I should probably not use just the ID for hierarchical data, and concatenate it with ancestors' ids, which would make keys unique.

This is the issue I just ran into, if this could be added in such a way that it allows for such subdivision without clashing keys (much like I assume other stacked displays do), that'd be excellent.
